builtin/common/vector.lua

```diff
@@ -131,3 +131,74 @@ function vector.divide(a, b)
 	end
 end
 
+-- returns a new normalized 2d vector with random values
+function vector.random2d()
+	local vect
+	repeat
+		vect = {x=math.random()*2-1, y=0, z=math.random()*2-1}
+	until vector.length(vect) <= 1
+	return vector.normalize(vect)
+end
+
+-- returns a new normalized 3d vector with random values
+function vector.random3d()
+	local vect
+	repeat
+		vect = {x=math.random()*2-1, y=math.random()*2-1, z=math.random()*2-1}
+	until vector.length(vect) <= 1
+	return vector.normalize(vect)
+end
+
+-- returns a vector with a specific length
+function vector.set_length(v, l)
+	v = vector.normalize(v)
+	return vector.multiply(v, l)
```

Comment:
@HybridDog  Please explain your change. What is wrong with my function? 0 * 0 = 0, so the math is correct, even for vectors. It's no division by 0. If the function returns something different the result may be unpredictable